### PR TITLE
User storage & subscription

### PR DESCRIPTION
**Pull request by BeaterKof**

### CODE REVIEW

File: file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/entity/Subscription.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        if (o == null || getClass() != o.getClass()) return false;
+        Subscription that = (Subscription) o;
+        return Objects.equals(id, that.id) &&
+                type == that.type &&
+                Objects.equals(storageSize, that.storageSize);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, type, storageSize);
```
Comment (Slidem) [inline file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/entity/Subscription.java]: We should be very carefull when choosing the identity of an jpa entity:
So here is a scenario:
- I have an entity with id 1
- I want to UPDATE this entity
- I want to update it with different storage size
- I create a new entity with id 1 and different storage size;
- Hibernate will try to create a NEW entity (insert a new row in the db instead of updating the one with id 1) , because the hash code differs
- I get a unique constrain error on the id

In conclusion, we should take care when defining hashcodes for enitities.
This is a good read : https://javaeeblog.wordpress.com/2017/11/15/identity-and-equality-of-jpa-entities/
In our case, I think we can use the ID hashcode generation.

File: file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/UserStorageService.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.file.sharing.core.entity.UserStorage;
+
+/**
+ * created by: andrei
+ * date: 24.03.2019
+ **/
+public interface UserStorageService {
+
+    public void createUserStorageByUserId(Integer userId);
```
Comment (Slidem) [inline file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/UserStorageService.java]: We should return the created storage object to be used by the client in other logic; just a good practice

File: file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -44,4 +45,10 @@ public UserInfo getUserInfoByUserId(Integer userId) {
 		}
 		return userInfo.get();
 	}
+
+	@Override
+	public Subscription getUserSubscription(int userId) {
+		Optional<UserInfo> userInfo = userBusiness.getUserInfoByUserId(userId);
+		return userInfo.map(ui -> ui.getAccStatsInfo().getSubscription()).orElseThrow(() -> new UserNotFoundException(userId));
```
Comment (Slidem) [inline file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/impl/UserServiceImpl.java]: This is a bit strange... If the Subscription in the AccStatsInfo of the UserInfo is null, I throw an exception UserNotFoundException. We should throw this exception only if the user info is not found. If AccStatsInfo is not nullable, and also the subcription is not nullable, we dont' need to do any other check.
Code would look something like this:

`
Optional<UserInfo> userInfo = userBusiness.getUserInfoByUserId(userId);

if( ! userInfo.isPresnet()){
    throw UserNotFoundException();
}

return userInfo.map(ui -> ui.getAccStatsInfo().getSubscription()).orElseThrow(() -> 
SubscriptionNotFoundException());
`

**Review by Slidem — CHANGES_REQUESTED** [review]
